### PR TITLE
Added canary url and stepfunction monitoring with sns email and slack…

### PR DESCRIPTION
**Pull request by bshien**
… integration

### Description

Creates a Cloudwatch Synthetic Canary to monitor the metrics.opensearch.org url and monitors the step functions for failure. If either of these fail, a message will be published to SNS, and will send email and slack notifications.

### Issues Resolved

https://github.com/opensearch-project/opensearch-metrics/issues/37

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: infrastructure/lib/constructs/snsMonitor.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        super(scope, id);
+        this.region = props.region;
+        this.accountId = props.accountId;
+        this.alarmNameSpace = props.alarmNameSpace;
+        this.snsTopicName = props.snsTopicName;
+        this.slackLambda = props.slackLambda;
+
+        // The email list for receiving alerts
+        this.emailList = [
+            'insert@mail.here'
```
Comment (bshien) [inline infrastructure/lib/constructs/snsMonitor.ts:36]: Move to project.ts

File: infrastructure/lib/stacks/secrets.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { Construct } from 'constructs';
+import {Secret} from "aws-cdk-lib/aws-secretsmanager";
+
+export class OpenSearchMetricsSecrets extends Stack {
+    readonly secretsObject: Secret;
+
+    constructor(scope: Construct, id: string) {
+        super(scope, id);
+        this.secretsObject = new Secret(this, 'MetricsCreds', {
+            secretName: 'metrics-creds',
```
Comment (bshien) [inline infrastructure/lib/stacks/secrets.ts:11]: input from project.ts

File: infrastructure/lib/stacks/secrets.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import {Stack} from "aws-cdk-lib";
+import { Construct } from 'constructs';
+import {Secret} from "aws-cdk-lib/aws-secretsmanager";
+
+export class OpenSearchMetricsSecrets extends Stack {
+    readonly secretsObject: Secret;
```
Comment (bshien) [inline infrastructure/lib/stacks/secrets.ts:6]: name it secret

File: src/main/java/org/opensearchmetrics/datasource/DataSourceType.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package org.opensearchmetrics.datasource;
+
+public enum DataSourceType {
+    SLACK_WEBHOOK_URL,
```
Comment (bshien) [inline src/main/java/org/opensearchmetrics/datasource/DataSourceType.java:4]: add unit tests

File: src/main/java/org/opensearchmetrics/lambda/SlackLambda.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    @Data
+    private static class AlarmObject {
+        @JsonProperty("AlarmName")
+        private String alarmName;
+        @JsonProperty("AlarmDescription")
+        private String alarmDescription;
```
Comment (bshien) [inline src/main/java/org/opensearchmetrics/lambda/SlackLambda.java:50]: move to model

File: src/main/java/org/opensearchmetrics/lambda/SlackLambda.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        @JsonProperty("StateChangeTime")
+        private String stateChangeTime;
+        @JsonProperty("Region")
+        private String region;
+        @JsonProperty("AlarmArn")
+        private String alarmArn;
+    }
```
Comment (prudhvigodithi) [inline src/main/java/org/opensearchmetrics/lambda/SlackLambda.java:57]: Move this `AlarmObject` to model/alarm package.

File: build.gradle [hunk 7]
```diff
@@ -28,6 +28,7 @@ dependencies {
     implementation 'io.github.acm19:aws-request-signing-apache-interceptor:2.3.1'
 
     implementation 'com.amazonaws:aws-lambda-java-core:1.2.3'
+    implementation 'com.amazonaws:aws-lambda-java-events:3.7.0'
```
Comment (prudhvigodithi) [inline build.gradle:31]: The latest one is `[3.11.5](https://mvnrepository.com/artifact/com.amazonaws/aws-lambda-java-events/3.11.5)`, can you please update ?

**Comment by prudhvigodithi** [comment]
LGTM, thanks @bshien.

**Pull request merged**